runtime/src/serde_snapshot.rs

```diff
@@ -74,7 +75,7 @@ const MAX_STREAM_SIZE: u64 = 32 * 1024 * 1024 * 1024;
 #[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
 #[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
 pub struct AccountsDbFields<T>(
-    HashMap<Slot, Vec<T>>,
+    HashMap<Slot, SmallVec<[T; 1]>>,
```

Comment:
ideally we could have something like `SingleSerializedAsSeq<T>` here, but I had trouble implementing such custom-serialized wrapper - small vec seems like an easier approach

However in the profile most of the time is actually taken by hashmap deserialization, so I think we might want to look at some kind of map <-> Vec<(K,V)> serialization util.